### PR TITLE
Standardization pass through docs/ folder. 

### PR DESCRIPTION
**Pull request by ScottTodd**
* recommend an out-of-tree CMake build directory `../iree-build/` instead of in-tree `build/`
* consistently use `.vmfb` as VM module flatbuffer file extensions
* make some docs use CMake examples instead of Bazel examples
* prefix all shell commands with `$`
* indent shell commands two spaces when continuing onto multiple lines
* consistently use `-` for LLVM flags and `--` for Abseil flags
* typo fixes

### CODE REVIEW

**Review by benvanik — APPROVED** [review]

**Comment by GMNGeoffrey** [comment]
I so badly want copy-paste that strips the `$`. We should look at how that's implemented internally and see if we can easily add it.

File: docs/get_started/getting_started_macos_bazel.md [hunk 1]
```diff
@@ -89,11 +89,11 @@ by git. The recommended contents for Linux/macOS are:
...
 build --disk_cache=/tmp/bazel-cache
 
-# Use --config=debug to compile iree and llvm without optimizations
+# Use --config=debug to compile IREE and LLVM without optimizations
 # and with assertions enabled.
 build:debug --config=asserts --compilation_mode=opt '--per_file_copt=iree|llvm@-O0' --strip=never
 
-# Use --config=asserts to enable assertions in iree and llvm.
+# Use --config=asserts to enable assertions in IREE and LLVM.
 build:asserts --compilation_mode=opt '--per_file_copt=iree|llvm@-UNDEBUG'
```
Comment (GMNGeoffrey) [inline docs/get_started/getting_started_macos_bazel.md:97]: Pretty sure @bjacob said this would create ODR violations and was bad

Comment (ScottTodd) [reply docs/get_started/getting_started_macos_bazel.md:97]: We can follow up on that later, separate from this PR.

Comment (GMNGeoffrey) [reply docs/get_started/getting_started_macos_bazel.md:97]: Yeah just noticed it :-)

**Comment by ScottTodd** [comment]
> I so badly want copy-paste that strips the `$`. We should look at how that's implemented internally and see if we can easily add it.

Same... Discussed a bit on Discord. We have some options, none look super easy though.

**Pull request merged**